scripts/tuple_tree_generator/tuple_tree_generator/generators/python.py

```diff
@@ -79,6 +94,114 @@ def _python_type(cls, resolved_type: Definition):
+            real_type = self.scalar_converter(real_type)
+
+        return real_type
+
+    def type_hint(self, field: StructField) -> str:
+        real_type = self._real_type(field)
+        possible_values = None
+        if real_type in BUILTINS_CLASSES:
+            hint_type = BUILTINS_CLASSES[real_type]
+            ctor = "native"
+        elif real_type in [e.name for e in self.schema.enum_definitions()]:
+            hint_type = "str"
+            ctor = "enum"
+            possible_values = f"{real_type}"
+        elif isinstance(field.resolved_type, ReferenceDefinition):
+            hint_type = "Reference"
+            ctor = "parse"
+        elif real_type in [
+            *(s.name for s in self.schema.struct_definitions() if s.abstract),
+            *(self.string_types),
+        ]:
+            hint_type = real_type
+            ctor = "parse"
+        else:
+            hint_type = real_type
+            ctor = "class"
+
+        assert hint_type
+
+        is_sequence = isinstance(field.resolved_type, SequenceDefinition)
+        if isinstance(field.resolved_type, SequenceDefinition):
+            is_abstract = isinstance(field.resolved_type.element_type, UpcastableDefinition)
+        else:
+            is_abstract = isinstance(field.resolved_type, UpcastableDefinition)
+
+        return (
+            f'{{"type": {hint_type}, '
+            + (f'"possibleValues": {possible_values},' if possible_values is not None else "")
+            + f'"ctor": "{ctor}", '
+            + f'"optional": {self.to_bool(field.optional)}, '
+            + f'"isArray": {self.to_bool(is_sequence)}, '
+            + f'"isAbstract": {self.to_bool(is_abstract)}}}'
+        )
+
+    def to_bool(self, input: bool):  # noqa: A002
+        if input:
+            return "True"
+        return "False"
+
+    def get_mixins(self, struct_name: str):
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                mixins_parsed = ast.parse(mixins.read())
+                mixin_classes = [
+                    node for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+
+                for mc in mixin_classes:
+                    if struct_name == mc.name.removesuffix("Mixin"):
+                        return mc.name
+        return "AllMixin"
+
+    def get_mixins_imports(self):
+        imports = []
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixins:
+                module_name = mixins_path.split("/")[-1]
+                mixins_parsed = ast.parse(mixins.read())
+                module_classes = [
+                    node.name for node in mixins_parsed.body if isinstance(node, ast.ClassDef)
+                ]
+                imports.append(
+                    f"from revng.model.{module_name.removesuffix('.py')} import {', '.join(module_classes)}"  # noqa: E501
```

Comment:
No special-casing the model, calculate a relative import (e.g. `from ..mixins import ...`) from the generated output.
Note: use as a target `"${CMAKE_BINARY_DIR}/python/revng/model/mixins.py"`, it's copied over by CMake

Reply:
Is [this](https://github.com/revng/revng/pull/398/commits/b0573c7b2b0d269a26c94bf589a249a071c2ea12#diff-452939ffa438785b15e5d75c947bcba221f52c898ee6f829b54da53157e40024R198) solution ok?

Reply:
Please don't link commits, those get deleted as soon as you push. If you have to show me something specific paste the needed snippet in the comment.
Anyways the implementation is fragile, see [here](https://github.com/revng/revng/pull/398#discussion_r1760984908) on how to make it more robust